game-core/src/main/java/games/strategy/engine/framework/message/PlayerListing.java

```diff
@@ -44,13 +47,25 @@
   /**
    * Creates a new instance of PlayerListing.
    */
-  public PlayerListing(final Map<String, String> playerToNodeListing, final Map<String, Boolean> playersEnabledListing,
-      final Map<String, PlayerType> localPlayerTypes, final Version gameVersion, final String gameName,
-      final String gameRound, final Collection<String> playersAllowedToBeDisabled,
+  public PlayerListing(
+      final Map<String, String> playerToNodeListing,
+      final Map<String, Boolean> playersEnabledListing,
+      final Map<String, PlayerType> localPlayerTypes,
+      final Version gameVersion,
+      final String gameName,
+      final String gameRound,
+      final Collection<String> playersAllowedToBeDisabled,
       final Map<String, Collection<String>> playerNamesAndAlliancesInTurnOrder) {
 
-    m_playerToNodeListing = Optional.ofNullable(playerToNodeListing).orElse(Collections.emptyMap());
-    m_playersEnabledListing = Optional.ofNullable(playersEnabledListing).orElse(Collections.emptyMap());
+    // Note: Sets from guava immutables are not necessarily serializable (!)
+    // We use copy constructors here to avoid this problem.
+
+    m_playerToNodeListing = Optional.ofNullable(playerToNodeListing)
+        .map(HashMap::new)
+        .orElse(new HashMap<>(0));
```

Comment:
Might want to lazily create these empty collections using `orElseGet()`.

Also, you could just use `Collections#emptyMap()` if you explicitly specify the type arguments to `Option#map()`.  It's a toss-up as to which is uglier: the explicit type arguments or having to `new` up an empty collection.  For example:

```diff
diff --git a/game-core/src/main/java/games/strategy/engine/framework/message/PlayerListing.java b/game-core/src/main/java/games/strategy/engine/framework/message/PlayerListing.java
index cc0c16ce2..112833240 100644
--- a/game-core/src/main/java/games/strategy/engine/framework/message/PlayerListing.java
+++ b/game-core/src/main/java/games/strategy/engine/framework/message/PlayerListing.java
@@ -61,26 +61,26 @@ public class PlayerListing implements Serializable {
     // We use copy constructors here to avoid this problem.
 
     m_playerToNodeListing = Optional.ofNullable(playerToNodeListing)
-        .map(HashMap::new)
-        .orElse(new HashMap<>(0));
+        .<Map<String, String>>map(HashMap::new)
+        .orElseGet(Collections::emptyMap);
     m_playersEnabledListing = Optional.ofNullable(playersEnabledListing)
-        .map(HashMap::new)
-        .orElse(new HashMap<>(0));
+        .<Map<String, Boolean>>map(HashMap::new)
+        .orElseGet(Collections::emptyMap);
     m_localPlayerTypes = Optional.ofNullable(localPlayerTypes.entrySet()
         .stream()
         // convert Map<String,PlayerType> -> Map<String,String>
         .collect(Collectors.toMap(Entry::getKey, e -> e.getValue().getLabel())))
-        .orElse(Collections.emptyMap());
+        .orElseGet(Collections::emptyMap);
     m_gameVersion = gameVersion;
     m_gameName = gameName;
     m_gameRound = gameRound;
     m_playersAllowedToBeDisabled = Optional.ofNullable(playersAllowedToBeDisabled)
-        .map(HashSet::new)
-        .orElse(new HashSet<>(0));
+        .<Collection<String>>map(HashSet::new)
+        .orElseGet(Collections::emptySet);
 
     m_playerNamesAndAlliancesInTurnOrder =
         Optional.ofNullable(playerNamesAndAlliancesInTurnOrder)
-            .orElse(Collections.emptyMap())
+            .orElseGet(Collections::emptyMap)
             .entrySet()
             .stream()
             .collect(Collectors.toMap(Entry::getKey, e -> Sets.newHashSet(e.getValue())));
```

Reply:
> Might want to lazily create these empty collections using orElseGet().

Whether there is a reference created to a supplier object, or an empty hashmap object created, it's a toss-up. Objects are dirt cheap to create. Looking at [HashMap](http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/util/HashMap.java) constructor source, I did not spot anything too expensive.

To another extent, it's a bit curious why one should not just always prefer: `orElseGet`? 

*********

re: type hinting vs new hashMap. 

Thanks for the pointer! It did not occur to me to try type hinting, I was trying a few other variants. You're right it's a trade-off of which coupling one wants, whether `Map` vs `HashMap` or `<>` vs `<...types...>`. The former is I think more good code hygiene, the latter a bit more practical as it is more likely to impact refactoring. I think it's a good bit more likely we'll need to change the type of map compared to changing from one map implementation to another.

Reply:
Good find on this problem in general, thanks for review and confirming fix @ssoloff .

Wrapping up:
- I updated this branch for some lazy loading: 6db447c
- Will keep the current `HashMap::new` over diamond notation type hinting